src/chatmessage.h

```diff
@@ -46,4 +47,20 @@ struct ChatMessage
 	std::wstring message = L"";
 	std::wstring sender = L"";
 	std::time_t timestamp = std::time(0);
+
+	std::string getType() const {
+		switch (type) {
+		case CHATMESSAGE_TYPE_RAW:
+			return "raw";
+		case CHATMESSAGE_TYPE_NORMAL:
+			return "normal";
+		case CHATMESSAGE_TYPE_ANNOUNCE:
+			return "announce";
+		case CHATMESSAGE_TYPE_SYSTEM:
+			return "system";
+		default:
+			FATAL_ERROR("Unexpected ChatMessageType error");
```

Comment:
It would be nice if we could enable the unhandled switch case warning and make it an error. It allows compile checking for this stuff.

I also don't like the `ENUM_MAX` paradigm :/

Reply:
You could put it into a string-enum lookup table, like commonly used in `script/`.